### PR TITLE
Cosmos DB: Adding yaml for databricks

### LINKED ISSUE #18868: CI to deploy to Databricks

### PR DESCRIPTION
**Pull request by ealsur**
Adds Yaml to trigger databricks integration.

Closes #18868

### CODE REVIEW

File: sdk/cosmos/ci.spark.databricks.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        $DATABRICKSENDPOINT
+
+        $DATABRICKSTOKEN
+
+        EOF
+  - task: Bash@3
+    displayName: Importing Jars
+    inputs:
+      filePath: databricks/databricks-jar-install.sh
+      arguments: '$(DatabricksClusterName) $(build.sourcesdirectory)/sdk/cosmos/azure-cosmos-spark_3-1_2-12/target azure-cosmos-spark_3-1_2-12-4.0.0-beta.2.jar '
```
Comment (moderakh) [inline sdk/cosmos/ci.spark.databricks.yml:48]: as this is hard coding the jar version in the CI, on each release we will have to update the Ci as well.

can we find a way to not hard code the version? a few options I can think of
1) auto discover the jar in the `target` folder,  the jar name format is `azure-cosmos-spark_3-1_2-12-4.0.0-${version}.jar '
2) use pom file to infer the version

File: sdk/cosmos/ci.spark.databricks.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+jobs:
+- job: Spark_CosmosDB_Databricks
```
Comment (moderakh) [inline sdk/cosmos/ci.spark.databricks.yml:2]: once this merge how this CI can be triggered?

is there a command for that?

File: sdk/cosmos/ci.spark.databricks.yml (hunk 1 above)
Comment (ealsur) [reply sdk/cosmos/ci.spark.databricks.yml:48]: Done, added auto detection

File: sdk/cosmos/ci.spark.databricks.yml (hunk 2 above)
Comment (ealsur) [reply sdk/cosmos/ci.spark.databricks.yml:2]: We should be able to trigger it with `/azp run <some-name>`

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/databricks/databricks-jar-install.sh [hunk 3]
```diff
@@ -2,10 +2,8 @@
 
 CLUSTER_NAME=$1
 JARPATH=$2
-JARFILE=$3
 [[ -z "$CLUSTER_NAME" ]] && exit 1
-[[ -z "$JARFILE" ]] && exit 1
-[[ -z "$JARFILE" ]] && exit 1
+[[ -z "$JARPATH" ]] && exit 1
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/databricks/databricks-jar-install.sh:6]: btw, could you please also rename the folder `azure-cosmos-spark_3-1_2-12/databricks`

I suspect end users may see this and may think it is a folder for sample code. how about renaming the folder to
`test-databricks` or something like that?

**Review by moderakh — APPROVED** [review]
minor comments, other than that LGTM.

File: sdk/cosmos/tests.yml [hunk 4]
```diff
@@ -49,3 +49,11 @@ stages:
       TestStepMavenInputs:
         goals: clean verify
         options: '$(ProfileFlag) -Dgpg.skip -Dmaven.wagon.http.pool=false $(DefaultOptions) -Dmaven.javadoc.skip=true -Drevapi.skip=true -pl $(ProjectList)'
+
+  - ${{ if or(eq(variables['Build.Reason'], 'Manual'), eq(variables['Build.Reason'], 'PullRequest')) }}:
+    - template: /sdk/cosmos/ci.spark.databricks.yml
+    parameters:
```
Comment (benbp) [inline sdk/cosmos/tests.yml:55]: Missing an indent here

Comment (ealsur) [reply sdk/cosmos/tests.yml:55]: Good catch

**Comment by benbp** [comment]
Template changes LGTM

**Pull request merged**